code/modules/jobs/job_types/_job.dm

```diff
@@ -531,11 +530,18 @@
 	if(!player_client)
 		return // Disconnected while checking for the appearance ban.
 
-	var/require_human = CONFIG_GET(flag/enforce_human_authority) && (job.job_flags & JOB_HEAD_OF_STAFF)
-	if(require_human)
-		var/all_authority_require_human = CONFIG_GET(flag/enforce_human_authority_on_everyone)
-		if(!all_authority_require_human && job.ignore_human_authority)
-			require_human = FALSE
+	var/human_authority_setting = CONFIG_GET(number/human_authority)
+
+	var/require_human = FALSE
+
+	if(job.job_flags & JOB_HEAD_OF_STAFF)
+		switch(human_authority_setting)
+			if(1)
+				require_human = job.human_authority == JOB_AUTHORITY_HUMANS_ONLY
+			if(2)
+				require_human = job.human_authority != JOB_AUTHORITY_NON_HUMANS_ALLOWED
+			if(3)
```

Comment:
please use defines for this, also what the fuck what is going on here

Reply:
I've added both defines and comments, hopefully its clearer now